core/src/main/java/org/apache/calcite/sql/SqlCollation.java

```diff
@@ -200,41 +200,29 @@ protected static SqlCollation getCoercibilityDyadic(
     case COERCIBLE:
       switch (coercibility2) {
       case COERCIBLE:
-        return new SqlCollation(
```

Comment:
@julianhyde , I don't quite get `new SqlCollation` thing here, however it looks like `new` here makes no difference. It might be `SqlCollation#strength` should be updated as well, however it does not seem to be supported in Calcite yet.

This change was required so `SqlCollation.IMPLICIT` singleton is propagated across all expressions.